Documentation/ryujit-tutorial.md

```diff
@@ -0,0 +1,728 @@
+# RyuJIT Tutorial
+
+## .NET Dynamic Code Execution
+
+### .NET Runtime
+- An implementation of the Common Language Infrastructure [ECMA 335]
+  - Supports multiple languages, including C#, F# and VB
+- Sources are mostly shared between the “desktop” version and the open-source coreclr implementation:
+    http://www.github.com/dotnet/coreclr
+- RyuJIT is the “next generation” just in time compiler for .NET
+
+#### Notes
+For context, the .NET runtime has been around since about the turn of the millennium. It a virtual machine that supports the execution of a number of languages, primarily C#, Visual Basic, and F#.
+It has been standardized through Ecma and then ISO.
+There have been a number of implementations of the CLI, the dominant one being installed with the Windows operating system.
+That implementation has served as the basis for a standalone implementation that is now available in open source.
+RyuJIT is the re-architected JIT for .NET.
+
+### Why “RyuJIT”?
+- Ryujin is a Japanese Sea Dragon
+
+We came up with the code name “RyuJIT” because our original code name had possible issues.
+We wanted something with “JIT” in the name, and the idea of a dragon came to mind because of the Dragon book that we all know and love.
+So – we just adapted the name of the Japanese sea dragon, Ryujin.
+
+### History
```

Comment:
Etymology of project name is a very fine story - there are a lots of pictures with Ryujins which are quite similar to a dragon on Dragon Book cover :)